raiden/network/rpc/client.py

```diff
@@ -456,6 +456,10 @@ def block_number(self):
         """ Return the most recent block. """
         return self.web3.eth.blockNumber
 
+    def blockhash(self):
+        """ Return the most recent block hash """
+        return self.blockhash_from_blocknumber(self.block_number())
```

Comment:
Please don't write code like this. Look at what this is doing:

This is an RPC call:

https://github.com/raiden-network/raiden/blob/e1b07d3c97e92946f84d01ef6e02a15bb6b044ac/raiden/network/rpc/client.py#L455-L457

Which returns an int that is then used on a next RPC call:

https://github.com/raiden-network/raiden/blob/e1b07d3c97e92946f84d01ef6e02a15bb6b044ac/raiden/network/rpc/client.py#L475-L478

https://github.com/raiden-network/raiden/blob/e1b07d3c97e92946f84d01ef6e02a15bb6b044ac/raiden/network/rpc/client.py#L463-L465

Please, just don't introduce yet another wrapper on `getBlock`, calling `get_block` is straight enough and this duplication of RPC calls has cropped up way too many times.

Reply:
The point is to not introduce unecessary RPC calls. You can get the exact same result by `blockhash_from_blocknumber('latest')`

Essentially getting the latest block and reading its blockhash.

Reply:
I see I can save RPC calls.

Shall I write

> blockhash_from_blocknumber('latest')

or

> get_block('latest')["blockNumber"]

in the body of this function or at the calling places?

Reply:
An argument for not adding the function would be to remove the clutter.

An argument for adding the function would be to avoid double nested RPC calls in the rest of the codebase.

Reply:
I think this function can disappear now.  Instead, I'm going to use `get_confirmed_blockhash()`.

Reply:
Probably the worst thing here was I was aware that I was making two RPC calls.  I thought about what can go wrong between the two RPC calls, and still kept them.

Reply:
>  I thought about what can go wrong between the two RPC calls, and still kept them.

Please refer to the issues about the number of RPC calls that are being done by Raiden: https://github.com/raiden-network/raiden/issues/4872